### PR TITLE
added cog type from pystac

### PR DESCRIPTION
**Pull request by emmanuelmathot**
**Proposed Changes:**

***Adding pystac COG type***
in [pystac](https://github.com/stac-utils/pystac/blob/develop/pystac/media_type.py#L10) COG is defined as `image/tiff; profile=cloud-optimized; application=geotiff` and thus not recognized in rio-tiler STAC reader. the COG type defined in pystac 0.5.4 is then added for compatibility.

**PR Checklist:**

- [X] This PR has **no** breaking changes.
- [X] I have added my changes to the [CHANGELOG](https://github.com/cogeotiff/rio-tiler/blob/master/CHANGES.md)

### CODE REVIEW

**Comment by emmanuelmathot** [comment]
The best way should be to analyze properly the content-type matching the media-type ans all the parameters (profile and application). Here I added the possible types encountered and the one defined in pystac in case it evolves in the future.

**Review by geospatial-jeff — APPROVED** [review]

**Comment by vincentsarago** [comment]
@emmanuelmathot thanks for the PR, do we have to add more tests?

**Review by kylebarron — APPROVED** [review]

File: rio_tiler/io/stac.py [hunk 1]
```diff
@@ -19,6 +19,8 @@
 DEFAULT_VALID_TYPE = {
     "image/tiff; application=geotiff",
     "image/tiff; application=geotiff; profile=cloud-optimized",
+    "image/tiff; profile=cloud-optimized; application=geotiff",
+    pystac.MediaType.COG,
```
Comment (kylebarron) [inline rio_tiler/io/stac.py:23]: Wait why are you adding both here? Aren't they identical?

Comment (emmanuelmathot) [reply rio_tiler/io/stac.py:23]: yes, they are but this ensures the compatibility even it changes in a future version of pystac.

Comment (vincentsarago) [inline rio_tiler/io/stac.py:23]: Would it be cleaner to do `pystac.MediaType.COG.value` ?

**Comment by vincentsarago** [comment]
@emmanuelmathot is this ready? I'm planning a release today!

File: CHANGES.md [hunk 2]
```diff
@@ -1,8 +1,10 @@
-## 2.0.4 (TBD)
+## 2.0.4 (2021-03-09)
 
+* Added [pystac.MediaType.COG](https://github.com/stac-utils/pystac/blob/master/pystac/media_type.py#L4) in supported types by STAC reader
 * fix bad type definition in `rio_tiler.colormap.ColorMaps` data (https://github.com/cogeotiff/rio-tiler/issues/359)
 * add `rio_tiler.colormap.parse_color` function to parse HEX color (https://github.com/cogeotiff/rio-tiler/issues/361)
 
+
```
Comment (vincentsarago) [inline CHANGES.md:7]:
```suggestion
```

File: rio_tiler/io/stac.py (hunk 1 above)
Comment (vincentsarago) [inline rio_tiler/io/stac.py:23]:
```suggestion
     pystac.MediaType.COG.value,
```

**Pull request merged**